src/components/TimeseriesChart/TimeseriesChart.test.tsx

```diff
@@ -104,3 +104,23 @@ describe('TimeseriesChart', () => {
     expect(wrapper.find('.context-container').exists()).toBeTruthy();
   });
 });
+
+it('renders with ruler specified', async () => {
+  const props = {
+    timeseriesIds: [timeseriesListV2[0].id],
+    ruler: {
+      visible: true,
+      yLabel: (point: ChartRulerPoint) => `${Number(point.value).toFixed(3)}`,
+      timeLabel: (point: ChartRulerPoint) =>
+        new Date(point.timestamp).toISOString(),
+    },
+  };
+  const wrapper = mount(
+    <ClientSDKProvider client={sdk}>
+      <TimeseriesChart {...props} />
+    </ClientSDKProvider>
+  );
+  await sleep(300);
+  wrapper.update();
+  expect(wrapper.find('.line').exists()).toBeTruthy();
```

Comment:
`expect(wrapper.find('.line').exists()).toBeTruthy();`
do you think this test is efficient enough? don't you need to look for cursor-overview instead?
you need to check if custom yLabel and timeLabel where called as well